lib/msal-browser/src/operatingcontext/BaseOperatingContext.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Logger } from "@azure/msal-common";
+import { BrowserConfiguration } from "../config/Configuration";
+
+/**
+ * Base class for operating context
+ * Operating contexts are contexts in which MSAL.js is being run
+ * More than one operating context may be available at a time
+ * It's important from a logging and telemetry point of view for us to be able to identify the operating context.
+ * For example: Some operating contexts will pre-cache tokens impacting performance telemetry
+ */
+export abstract class BaseOperatingContext {
+
+    protected logger:Logger;
+    protected config:BrowserConfiguration;
+    protected available:boolean;
+
+    constructor(logger:Logger, config:BrowserConfiguration){
+        this.config = config;
+        this.logger = logger;
+        this.available = false;
+    }
+
+    /**
+     * returns the name of the module containing the API controller associated with this operating context
+     */
+    abstract getModuleName():string;
+
+    /**
+     * returns the string identifier of this operating context
+     */
+    abstract getId():string;
+
+    /**
+     * returns a boolean indicating whether this operating context is present
+     */
+    abstract isAvailable():Promise<boolean>;
+
+    /**
+     * Return the MSAL config
+     * @returns BrowserConfiguration
+     */
+    getConfig():BrowserConfiguration {
+        return this.config;
+    }
+
+    /**
+     * Returns the MSAL Logger
+     * @returns Logger
+     */
+    getLogger():Logger {
+        return this.logger;
+    }
+
+    getAvailable():boolean {
```

Comment:
Difference between getAvailable and isAvailable?

Reply:
Looks like `isAvailable()` acts as both setter and getter, while `getAvailable()` is just a getter function.

Reply:
I'll take a look.

Reply:
getAvailable is sync and isAvailable is Async.  Do we have an approach that we're using for this pattern?

Reply:
Changed this to initialize and isAvailable

Reply:
fixed.  Hopefully more readable now.